src/pages/NotionPage.js

```diff
@@ -0,0 +1,242 @@
+import SidebarHeader from '../components/Sidebar/SidebarHeader.js';
+import DocumentList from '../components/Sidebar/DocumentList.js';
+import CreateButton from '../components/Sidebar/CreateButton.js';
+import DocumentHeader from '../components/Document/DocumentHeader.js';
+import Editor from '../components/Document/Editor.js';
+
+export default function NotionPage({ $target, initialState }) {
+  const $sidebarContainer = document.createElement('div');
+  const $documentContainer = document.createElement('div');
+  $sidebarContainer.classList.add('sidebar-container');
+  $documentContainer.classList.add('document-container');
+  $target.appendChild($sidebarContainer);
+
+  this.state = initialState;
+
+  let documentLocalSaveKey = `${TEMP_DOCUMENT_KEY(this.state.documentId)}`;
+  let timer = null;
+  const doc = getItem(documentLocalSaveKey, INIT_DOCUMENT);
+
+  new SidebarHeader({
+    $target: $sidebarContainer,
+    initialState: {
+      heading: '📚 Notion',
+    },
+  });
+
+  const documentList = new DocumentList({
+    $target: $sidebarContainer,
+    initialState: {
+      documents: this.state.documents,
+      documentId: this.state.documentId,
+      openedDocuments: this.state.openedDocuments,
+    },
+    onToggle: (id, $ul) => {
+      const index = this.state.openedDocuments.indexOf(parseInt(id));
+
+      if (index > -1) {
+        // 토글 닫기
+        const copyOpenedDocuments = [...this.state.openedDocuments];
+        copyOpenedDocuments.splice(index, 1);
+
+        closeSubList($ul, copyOpenedDocuments);
+
+        this.setState({
+          ...this.state,
+          openedDocuments: copyOpenedDocuments,
+        });
+        setItem(OPENED_DOCUMENTS_KEY, copyOpenedDocuments);
+      } else {
+        // 토글 열기
+        this.setState({
+          ...this.state,
+          openedDocuments: [...this.state.openedDocuments, parseInt(id)],
+        });
+        setItem(OPENED_DOCUMENTS_KEY, this.state.openedDocuments);
+      }
+    },
+    onCreate: async (id) => {
+      const createdDocument = await request('/documents', {
+        method: 'POST',
+        body: JSON.stringify({ title: '', parent: id }),
+      });
+      await fetchDocuments();
+
+      push(`/documents/${createdDocument.id}`);
+
+      this.setState({
+        ...this.state,
+        openedDocuments: [...this.state.openedDocuments, id],
+      });
+      setItem(OPENED_DOCUMENTS_KEY, this.state.openedDocuments);
+    },
+    onDelete: async (id) => {
+      await request(`/documents/${id}`, {
+        method: 'DELETE',
+      });
+      await fetchDocuments();
+
+      if (id === this.state.documentId) {
+        if (this.state.documents.length > 0) {
+          push(`/documents/${this.state.documents[0].id}`);
+        } else {
+          push('/');
+        }
+      } else {
+        if (this.state.documentId) {
+          push(`/documents/${this.state.documentId}`);
+        } else {
+          push('/');
+        }
+      }
```

Comment:
조건문의 분기 구조는 가능하면 최대한 flat하게 만들어주는 게 좋습니다.
더 나아가서는 `id === this.state.documentId && this.state.documents.length > 0`처럼 복잡한 조건 판단식의 경우는 함수로 추출해주면 분기 구조를 더욱 명확하게 파악할 수 있습니다.

```suggestion
      if (id === this.state.documentId && this.state.documents.length > 0) {
        push(`/documents/${this.state.documents[0].id}`);
      } else if (this.state.documentId) {
        push(`/documents/${this.state.documentId}`);
      } else {
        push('/');
      }
```